packages/common-all/src/git.ts

```diff
@@ -0,0 +1,242 @@
+import _ from "lodash";
+import path from "path";
+import type { DVault, DendronConfig, NoteProps } from "./types";
+import { RESERVED_KEYS, FOLDERS } from "./constants";
+import { ConfigUtils } from "./utils";
+import { VaultUtils } from "./vault";
+
+const formatString = (opts: { txt: string; note: NoteProps }) => {
+  const { txt, note } = opts;
+  _.templateSettings.interpolate = /{{([\s\S]+?)}}/g;
+  const noteHierarchy = note.fname.replace(/\./g, "/");
+  return _.template(txt)({ noteHierarchy });
+};
+
+/**
+ *  NOTICE: Lots of the Git code is obtained from https://github.com/KnisterPeter/vscode-github, licened under MIT
+ */
+
+/**
+ * Utilities for working with git urls
+ */
+
+export function canShowGitLink(opts: {
```

Comment:
can we add some tests for these methods?

Reply:
Bootstaped with `canShowGitLink` util https://github.com/dendronhq/dendron/pull/3681/commits/bfb2d56d5c982cbb2409839f04eda965d03fc94b

Shall I continue adding tests for the rest of the utils?

Reply:
Added a couple more but leave it now as it is [`0e6b42a` (#3681)](https://github.com/dendronhq/dendron/pull/3681/commits/0e6b42ab7f8297e0a46fd3421e1dbc37b6b3e739)